contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,91 @@
+# Deprecating and promoting components in Primer React
```

Comment:
I am not sure if "Promoting" a good word for describing the process to move the deprecated component into the deprecated bundle. I thought it would be good to separate the process of deprecation (as in adding a `@deprecated` annotation) and moving the component into the deprecated folder.

Reply:
cc @colebemis